system/lib/wasmfs/file.cpp

```diff
@@ -39,6 +40,16 @@ __wasi_errno_t MemoryFile::read(uint8_t* buf, size_t len, off_t offset) {
 
   return __WASI_ERRNO_SUCCESS;
 }
+
+void MemoryFile::Handle::preloadFromJS(int index, int fileSize) {
```

Comment:
Can we read the size out of JS here rather than in the caller? That way we would have a clear division: everything required to create the file is handled in `WasmFS::preloadFiles` and everything required to fill the file is handled here.